### DISCUSSION TITLE
How to feed MXNet with data already on GPU memory?

### DISCUSSION THREAD

**Discussion by kice** [apache/incubator-mxnet#13688] [Q&A]
I want to feed the image data already on GPU (in fact, it is a ID3D11Texture2D object) to MXNet; and if possiable, after processing, I can display it with DirectX. 

I think I can just use `cudaMemcpy` or something like that to make a bridge. Do anyone have any idea for this kind of feature?

**Comment by andrewfayres** [comment]
Adding some labels for visibility. I recommend asking this on discuss.mxnet.io. Questions tend to get a wider audience there.

@mxnet-label-bot add [question, cuda, windows]

**Comment by Ishitori** [comment]
I don't think it is possible, unless you are able to code your data into DLPack format: https://github.com/dmlc/dlpack In that case, if you could also provide a memory pointer to MXNet code, you could use [from_dlpack](https://mxnet.incubator.apache.org/api/python/ndarray/ndarray.html#mxnet.ndarray.from_dlpack) method to get the NDArray.
